src/apollo/utils/Secp256k1PrivateKey.ts

```diff
@@ -49,31 +64,32 @@ export class Secp256k1PrivateKey
     const seedHex = this.getProperty(KeyProperties.seed);
 
     if (!seedHex) {
-      throw new Error("Seed not found");
+      throw new ApolloError.MissingKeyParameters([KeyProperties.seed]);
     }
 
     const seed = Buffer.from(seedHex, "hex");
     const derivationPathStr = derivationPath.toString();
 
-    const newExtendedKey = HDKey.InitFromSeed(
+    const baseKey = HDKey.InitFromSeed(
       Int8Array.from(seed),
       0,
       BigIntegerWrapper.initFromInt(0)
-    ).derive(derivationPathStr);
-
-    const newExtendedPrivateKey = new Secp256k1PrivateKey(
-      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-      Uint8Array.from(newExtendedKey.privateKey!)
     );
+    const derivedKey = baseKey.derive(derivationPathStr);
+
+    if (derivedKey.privateKey == null) {
+      throw new ApolloError.MissingPrivateKey();
+    }
 
-    newExtendedPrivateKey.keySpecification.set(KeyProperties.seed, seedHex);
-    newExtendedPrivateKey.keySpecification.set(
-      KeyProperties.derivationPath,
-      Buffer.from(derivationPathStr).toString("hex")
+    const privateKey = new Secp256k1PrivateKey(
+      Uint8Array.from(derivedKey.privateKey)
     );
-    newExtendedPrivateKey.keySpecification.set(KeyProperties.index, "0");
 
-    return newExtendedPrivateKey;
+    privateKey.keySpecification.set(KeyProperties.seed, seedHex);
+    privateKey.keySpecification.set(KeyProperties.derivationPath, Buffer.from(derivationPathStr).toString("hex"));
```

Comment:
Yes like so this is acceptable. The seed should always be passed by the user no matter what.

The line before is not ;)